### PR TITLE
datasource/gce: add initial support

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

**Comment by crawford** [comment]
```
$ sudo ./coreos-cloudinit -oem=gce
2016/05/18 20:13:48 Checking availability of "gce-metadata-service"
2016/05/18 20:13:48 Fetching user-data from datasource of type "gce-metadata-service"
2016/05/18 20:13:48 Fetching data from http://metadata.google.internal/computeMetadata/v1/instance/attributes/user-data. Attempt #1
2016/05/18 20:13:48 Fetching meta-data from datasource of type "gce-metadata-service"
2016/05/18 20:13:48 Fetching data from http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/access-configs/0/external-ip. Attempt #1
2016/05/18 20:13:48 Fetching data from http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/ip. Attempt #1
2016/05/18 20:13:48 Fetching data from http://metadata.google.internal/computeMetadata/v1/instance/hostname. Attempt #1
2016/05/18 20:13:48 Merging cloud-config from meta-data and user-data
2016/05/18 20:13:48 Set hostname to crawford-ignition-test-1.c.coreos-gce-testing.internal
2016/05/18 20:13:48 Writing file to "/etc/environment"
2016/05/18 20:13:48 Wrote file to "/etc/environment"
2016/05/18 20:13:48 Updated /etc/environment
2016/05/18 20:13:48 Ensuring runtime unit file "etcd.service" is unmasked
2016/05/18 20:13:48 Ensuring runtime unit file "etcd2.service" is unmasked
2016/05/18 20:13:48 Ensuring runtime unit file "fleet.service" is unmasked
2016/05/18 20:13:48 Ensuring runtime unit file "locksmithd.service" is unmasked
```

```
$ cat /etc/environment 
COREOS_PRIVATE_IPV4=10.240.0.14
COREOS_PUBLIC_IPV4=104.196.121.100
```

File: datasource/metadata/gce/metadata.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+type metadataService struct {
+	metadata.MetadataService
+}
+
+func NewDatasource(root string) *metadataService {
+	return &metadataService{metadata.NewDatasource(root, apiVersion, userdataPath, metadataPath, http.Header{"Metadata-Flavor": {"Google"}})}
+}
+
+func (ms metadataService) FetchMetadata() (datasource.Metadata, error) {
+	public, err := ms.fetchIP("network-interfaces/0/access-configs/0/external-ip")
```
Comment (marineam) [inline datasource/metadata/gce/metadata.go]: What happens if the machine has no external IP?

Comment (marineam) [inline datasource/metadata/gce/metadata.go]: Hint: it shouldn't fail ;-)

Comment (crawford) [inline datasource/metadata/gce/metadata.go]: Now it doesn't!

**Comment by marineam** [comment]
LGTM

**Pull request merged**